tools/azure-pipelines/jobs-template.yml

```diff
@@ -160,14 +160,17 @@ jobs:
         echo "##vso[task.setvariable variable=PATH]$JAVA_HOME_11_X64/bin:$PATH"
       displayName: "Set to jdk11"
       condition: eq('${{parameters.jdk}}', 'jdk11')
-    - script: ./tools/travis/setup_maven.sh
+    - script: |
+        source ./tools/ci/maven-utils.sh
+        setup_maven
+        echo "##vso[task.setvariable variable=M2_HOME]$M2_HOME"
```

Comment:
are exports from `setup_maven` not visible in subsequent steps?

Reply:
no :(

Reply:
then we may want to explicitly setup maven in the azure_controller.yml / run-nightly-tests.sh files. Conceptually we _shouldn't_ as it is about setting up the environment, but having to "manually" pass exported variables seems like something we will eventually stumble over, and it already raises some questions.

For example, why are we only carrying over the `M2_HOME` variable, and not `PATH`?

Reply:
I agree that this is not optimal, but in my evaluation, this proposal seems to be the least ugliest approach.

I first considered adding the maven setup into the `run-nightly-tests.sh` script. But that would install maven in the user's home directory, when running the script locally.
The `azure_controller.sh` script is another option, yes. And in some sense is the azure_controller.sh an environment specific script.

> For example, why are we only carrying over the M2_HOME variable, and not PATH?

In my mental model, the contract of `run_mvn` is that is runs "mvn" from `PATH`, unless `M2_HOME` is set. If `M2_HOME` is set, it uses that mvn installation.
Exporting PATH as well would be redundant (I should actually remove the `export PATH=${M2_HOME}/bin:${PATH}` line from the `setup_maven` function.
How about doing that + adding a comment about the contract?

Reply:
> But that would install maven in the user's home directory, when running the script locally.

Fair point, but didn't you argue previously that no one should use `run-nightly-tests.sh` anyway since it installs so many things? I thought we relegated this script to only be used for azure.

> I should actually remove the export PATH=${M2_HOME}/bin:${PATH} line from the setup_maven function.

+1

Reply:
Okay, so how do we proceed here?

In any case, I'll remove the `export PATH`.

Option A) I move the maven setup to `run-nightly-tests.sh`, I would also move the file itself into the `ci/` directory and add a warning so that people don't run it on their machines.

Option B) I move the maven setup to the azure controller

Option C) We only remove the PATH export.

Reply:
There may be another option.

In another PR you added `echo "##vso[task.setvariable variable=ARTIFACT_DIR]$(pwd)/compressed-archive-dir"` to `run-nightly-test.sh`, which I presume exposes this variable in a way that subsequent steps can see it.
Could we not do the same for maven?

Reply:
True. So I just move the setvariable call into the setup method?

Reply:
yes. then we have all the export logic in one place.